src/libproc_macro/bridge/buffer.rs

```diff
@@ -139,12 +139,14 @@ impl<T: Copy> From<Vec<T>> for Buffer<T> {
             }
         }
 
+        #[allow(improper_ctypes)]
```

Comment:
The lint only complains about these functions because of the type parameter, right? Kind of silly (especially since it's apparently only ever used with `T = u8`) but AFAICT there's no way to make this more precise without deferring to monomorphization time, which would be feasible but risks showing the warning to the wrong people (downstream crates who didn't make any mistake).

Reply:
I've disabled the lint for generic functions, as per your second suggestion, see [this comment](https://github.com/rust-lang/rust/pull/65134#issuecomment-546625981) - I'll resolve the comments you've left that are due to generics.